File: AtCoder/ACP/BootCamp4b/Medium100_2/python/B_KEYENCE_String.py
S = input()
T = 'keyence'

NS = len(S)
NT = len(T)
for i in range(len(T) + 1):
    if S[:i] == T[:i] and S[NS - NT + i:] == T[i:]:
        print("YES")
        exit()
print("NO")

File: AtCoder/ACP/BootCamp4b/Medium100_2/cpp/B_KEYENCE_String.cpp
#include <bits/stdc++.h>
using namespace std;

const string T = "keyence";

string S;

void solve() {
  int NS = S.size();
  int NT = T.size();
  for (int i = 0; i <= NT; ++i) {
    if (S.substr(0, i) == T.substr(0, i) && S.substr(NS - NT + i) == T.substr(i)) {
      puts("YES");
      return;
    }
  }
  puts("NO");
}

int main() {
  cin >> S;
  solve();
}